pkg/lib/authn/authenticator/service/service.go

```diff
@@ -513,7 +514,11 @@ func (s *Service) verifyWithSpec(info *authenticator.Info, spec *authenticator.S
 		plainPassword := spec.Password.PlainPassword
 		a := info.Password
 		requireUpdate, err = s.Password.Authenticate(a, plainPassword)
-		if err != nil {
+		var apiError = apierrors.AsAPIError(err)
+		if apiError != nil && apiError.Kind == password.PasswordExpiryForceChange {
+			// Proceed next auth step but require update
+			err = api.ErrPasswordExpiryForceChange
```

Comment:
This pattern of reporting information is very easy to miss. It requires to the caller to pay attention that the error sometimes does not mean an error. 

Let's change the interface so that the caller must be aware of the possible outcome.

Originally: `(info *authenticator.Info, requireUpdate bool, err error)`
Now: `(info *authenticator.Info, result *VerifyResult, err error)`

```golang
type VerifyResult struct {
  PasswordPolicyForceChange bool
  PasswordExpiryForceChange bool
}
```

With this change, this PR should be greatly simplified. We just propagate VerifyResult and we can differentiate the case.

I also notice that we need to add a piece of data in the `change_password` step to tell the caller why the change is needed. The step is using `new_password_data`. Let's add a new data `force_change_password_data`. This new data contains a new field `reason`. The valid values for `reason` is `policy` or `expiry`.

Reply:
If that's the case seems better to propagate to workflow / interaction to decide what next action is?